Explorer/Assets/DCL/NameTags/NametagView.cs

```diff
@@ -141,44 +132,70 @@ public void SetChatMessage(string chatMessage)
             cts.SafeCancelAndDispose();
             cts = new CancellationTokenSource();
 
-            StartChatBubbleFlowAsync(chatMessage).Forget();
+            StartChatBubbleFlowAsync(chatMessage, cts.Token).Forget();
         }
 
-        private async UniTaskVoid StartChatBubbleFlowAsync(string chatMessage)
+        private void ResetElement()
         {
-            if (isAnimatingIn || isWaiting)
-                await AnimateOutAsync();
-
-            await AnimateInAsync(chatMessage);
-            isAnimatingIn = false;
-            isWaiting = true;
-            await UniTask.Delay(chatBubbleConfiguration.bubbleIdleTime + AdditionalMessageVisibilityTimeMs(chatMessage), cancellationToken: cts.Token);
-            isWaiting = false;
-            await AnimateOutAsync();
-            isAnimatingOut = false;
+            preferredSize = Vector2.zero;
+            backgroundFinalSize = Vector2.zero;
+            textContentInitialPosition = Vector2.zero;
+            usernamePos = Vector2.zero;
+            Username.text = string.Empty;
+            Username.rectTransform.anchoredPosition = Vector2.zero;
+            MessageContent.text = string.Empty;
+            Background.size = Vector2.zero;
+            previousDistance = 0;
+        }
+
+        private async UniTaskVoid StartChatBubbleFlowAsync(string chatMessage, CancellationToken ct)
+        {
+            try
+            {
+                if (isAnimatingIn || isWaiting)
+                    await AnimateOutAsync(ct);
+
+                await AnimateInAsync(chatMessage, ct);
+
+                isAnimatingIn = false;
+                isWaiting = true;
+
+                await UniTask.Delay((chatBubbleConfiguration?.bubbleIdleTime ?? 5000) + AdditionalMessageVisibilityTimeMs(chatMessage), cancellationToken: ct);
```

Comment:
Can you introduce a named constant instead of `5000`?

Reply:
Converted all default settings into constants